### DISCUSSION TITLE
How do I specify a custom color for a style? (Spectre.Console.Color)

### DISCUSSION THREAD

**Discussion by futuremotiondev** [spectreconsole/spectre.console#1270] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
No, the request is not related to a problem. Just trying to find out how to accomplish what I need.

**Describe the solution you'd like**
The ability to use a HTML hex code for a color, or an RGB value that I can feed into the `Spectre.Console.Style` constructor.

**Describe alternatives you've considered**
I have considered no alternatives. 

**Additional context**
I just want to specify a custom color. 

**Edit:**

I just want to clarify that I know you can use custom hex colors in Markup but there is no way that I can find to define custom hex colors in `Spectre.Console.Style`. I'm trying to make a `Spectre.Console.SelectionPrompt` and specify a custom color for the **highlighted / selected item**.

**Comment by patriksvensson** [comment, marked as answer]
## API docs

* https://spectreconsole.net/api/spectre.console/style/
* https://spectreconsole.net/api/spectre.console/color/

```
public Style(Color? foreground = null, Color? background = null, Decoration? decoration = null, string link = null)
```

## Example

```csharp
// Create a style with green foreground
var style = new Style(
    foreground: new Color(0, 0xFF, 0))
```

**Comment by mbianchetto** [comment]
Sorry for the stupid question.

            var style = new Style(foreground: new Color(0, 0xFF, 0));
            AnsiConsole.WriteLine("Hello", new Style(Color.Red, Color.Yellow, Decoration.Bold));  // does not work
            AnsiConsole.WriteLine("Hello", style);  // does not work
            AnsiConsole.Markup("[maroon on blue]Hello[/]"); // it works perfectly

Where am I going wrong?

**Reply by patriksvensson** [reply]
@mbianchetto There is no `WriteLine` override that takes a `Style` as an argument. You will need to render a `Markup` widget with the specified text:

```csharp
AnsiConsole.Write(new Markup("Hello", style));
AnsiConsole.WriteLine();
```

**Reply by mbianchetto** [reply]
My bad,I forgot about  RTFM thank you very much
